pkg/cmdutil/json_flags.go

```diff
@@ -0,0 +1,101 @@
+package cmdutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/cli/cli/pkg/export"
+	"github.com/cli/cli/pkg/jsoncolor"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type JSONFlagError struct {
+	error
+}
+
+func AddJSONFlags(cmd *cobra.Command, exportTarget **ExportFormat, fields []string) {
```

Comment:
It looks like we often want to use the `json` field passed in by the user or a set of default fields, what do you think about passing the default fields into this function and using them as the default for the `json` flag? If we did this we would probably also want to make sure that in the case when the `json` flag is not specified we don't return `nil` but rather a `ExportFormat` struct with some indication, maybe a new field, that states if a non-default format was specified.

Reply:
It's a good thought but I'd like to avoid specifying the default at the flag level because the flag serves two purposes at once: to opt into JSON export mode, and to control what data is fetched. If there was a default at the flag level, the reader might mis-interpret it that those default fields somehow pertain to the export, whereas they don't.

I'm fine for now passing default fields through a separate mechanism, because normal fetch represents a different mode of operation than the `--json` mode.